homeassistant/components/proxmoxve/__init__.py

```diff
@@ -86,93 +86,103 @@
         hass.data[PROXMOX_CLIENTS] = {}
 
-        for entry in config[DOMAIN]:
-            host = entry[CONF_HOST]
-            port = entry[CONF_PORT]
-            user = entry[CONF_USERNAME]
-            realm = entry[CONF_REALM]
-            password = entry[CONF_PASSWORD]
-            verify_ssl = entry[CONF_VERIFY_SSL]
+        host = entry_data[CONF_HOST]
+        port = entry_data[CONF_PORT]
+        user = entry_data[CONF_USERNAME]
+        realm = entry_data[CONF_REALM]
+        password = entry_data[CONF_PASSWORD]
+        verify_ssl = entry_data[CONF_VERIFY_SSL]
 
-            hass.data[PROXMOX_CLIENTS][host] = None
+        hass.data[PROXMOX_CLIENTS][host] = None
 
-            try:
-                # Construct an API client with the given data for the given host
-                proxmox_client = ProxmoxClient(
-                    host, port, user, realm, password, verify_ssl
-                )
-                proxmox_client.build_client()
-            except AuthenticationError:
-                _LOGGER.warning(
-                    "Invalid credentials for proxmox instance %s:%d", host, port
-                )
-                continue
-            except SSLError:
-                _LOGGER.error(
-                    "Unable to verify proxmox server SSL. "
-                    'Try using "verify_ssl: false" for proxmox instance %s:%d',
-                    host,
-                    port,
-                )
-                continue
-            except ConnectTimeout:
-                _LOGGER.warning("Connection to host %s timed out during setup", host)
-                continue
+        try:
+            # Construct an API client with the given data for the given host
+            proxmox_client = ProxmoxClient(
+                host, port, user, realm, password, verify_ssl
+            )
+            proxmox_client.build_client()
+        except AuthenticationError:
+            _LOGGER.warning(
+                "Invalid credentials for proxmox instance %s:%d", host, port
+            )
+            return False
+        except SSLError:
+            _LOGGER.error(
+                "Unable to verify proxmox server SSL. "
+                'Try using "verify_ssl: false" for proxmox instance %s:%d',
+                host,
+                port,
+            )
+            return False
+        except ConnectTimeout:
+            _LOGGER.warning("Connection to host %s timed out during setup", host)
+            return False
 
-            hass.data[PROXMOX_CLIENTS][host] = proxmox_client
+        hass.data[PROXMOX_CLIENTS][host] = proxmox_client
 
     await hass.async_add_executor_job(build_client)
 
     coordinators = hass.data[DOMAIN][COORDINATORS] = {}
 
-    # Create a coordinator for each vm/container
-    for host_config in config[DOMAIN]:
-        host_name = host_config["host"]
-        coordinators[host_name] = {}
+    host_name = entry_data["host"]
+    coordinators[host_name] = {}
```

Comment:
Each config entry already is unique for the host? Does this need to be namespace?

```suggestion
    coordinators = {}
```